### DISCUSSION TITLE
Forgejo: private submodules are not accessible

### DISCUSSION THREAD

**Discussion by mrwsl** [renovatebot/renovate#31785] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Forgejo; docker image renovate:latest

### Please tell us more about your question or problem

I followed Gitea's [blog entry](https://about.gitea.com/resources/tutorials/use-gitea-and-renovate-bot-to-automatically-monitor-software-packages#:~:text=This%20tutorial%20will%20show%20you%20how) to set up renovate on a self-hosted forgejo instance.

I struggle to get submodules that are also hosted on forgejo working:
```
DEBUG: Found hostRules token for url https://git.tld.com/mirror/benchmark.git (repository=organization/cpp_template_program)
WARN: Error mapping git submodules during extraction (repository=organization/cpp_template_program, packageFile=.gitmodules)
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--symref",
             "https://**redacted**@git.tld.com/organization/benchmark.git",
             "HEAD"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "remote: User permission denied\nfatal: unable to access 'https://git.tld.com/organization/benchmark.git/': The requested URL returned error: 403\n",
         "stack": "Error: remote: User permission denied\nfatal: unable to access 'https://**redacted**@3.27.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/renovate/node_modules/.pnpm/simple-git@3.27.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:54:29)\n    at /usr/local/renovate/node_modules/.pnpm/simple-git@3.27.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/renovate/node_modules/.pnpm/simple-git@3.27.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:121:14)\n    at GitExecutorChain.<anonymous> (/usr/local/renovate/node_modules/.pnpm/simple-git@3.27.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:97:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/renovate/node_modules/.pnpm/simple-git@3.27.0/node_modules/simple-git/dist/cjs/index.js:52:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }
```

I set up a hostRule for git-refs since checking the submodules was not working with the default config:
```
  hostRules: [
    {
      hostType: "git-refs",
      matchHost: "git.tld.com",
      token: process.env.RENOVATE_TOKEN,
    }
]
```
Funnily enough, I can run the `git ls-remote` command printed in the log on my system fine:
```
git ls-remote --symref https://token@git.tld.com/organization
/benchmark.git 
ref: refs/heads/main	HEAD
...
```

I found a [similar discussion](https://forum.gitea.com/t/renovate-permission-issues/8948) in the Gitea forum without a solution to the problem. What did I miss?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
Did you tried `git ls-remote https://<token>@git.tld.com/organization/benchmark.git/` ?

Your sample show a different url

**Reply by mrwsl** [reply]
Yes, this is the faulty result of my search and replace of private information. I tried `git ls-remote https://<token>@git.tld.com/organization/benchmark.git` locally successfully.

**Comment by iuliandita** [comment]
hey @mrwsl - were you able to solve this?

**Reply by mrwsl** [reply]
No luck yet :/
